src/main/scala/ing/wbaa/druid/DruidQuery.scala

```diff
@@ -63,18 +63,55 @@ case class GroupByQuery(
     granularity: Granularity = GranularityType.All,
     dataSource: String = DruidConfig.datasource,
     having: Option[Having] = None,
-    postAggregations: List[PostAggregation] = Nil
+    limitSpec: Option[LimitSpec] = None,
+    postAggregations: List[PostAggregation] = List()
 ) extends DruidQuery {
+
   val queryType = QueryType.GroupBy
 }
 
+case class LimitSpec(limit: Int, columns: Seq[OrderByColumnSpec]) {
+  val `type` = "default"
+}
+
+object LimitSpec {
+  implicit val encoder: Encoder[LimitSpec] = new Encoder[LimitSpec] {
+    override def apply(a: LimitSpec): Json = a.asJsonObject.add("type", a.`type`.asJson).asJson
+  }
+}
+
+case class OrderByColumnSpec(
+    dimension: String,
+    direction: Direction = Direction.ascending,
+    dimensionOrder: DimensionOrder = DimensionOrder()
+)
+case class DimensionOrder(`type`: DimensionOrderType = DimensionOrderType.lexicographic)
+
+sealed trait Direction extends Enum with LowerCaseEnumStringEncoder
+object Direction extends EnumCodec[Direction] {
+  case object ascending  extends Direction
```

Comment:
Is there a specific reason to not use a capital in defining the object? That would be `Scala` convention right?. Json serialization would take care of the lowercasing in this case

Reply:
I think there are other places where the same comment applies

Reply:
For some reason I haven't noticed that I didn't use capital when defining the object name in enumerations. You are right, Json serialization will convert it to lowercase.